xapian-core/include/xapian/cluster.h

```diff
@@ -28,14 +28,43 @@
 #endif
 
 #include <xapian/mset.h>
+#include <xapian/queryparser.h>
 #include <xapian/types.h>
 #include <xapian/visibility.h>
 
 #include <unordered_map>
+#include <unordered_set>
 #include <vector>
 
 namespace Xapian {
 
+/** Implementation of the stopper class that can be used to identify
+ *  stemmed as well as unstemmed stopwords
+ */
+class XAPIAN_VISIBILITY_DEFAULT StemStopper : public Xapian::Stopper {
+    std::unordered_set<std::string> stop_words;
+    Xapian::Stem stemmer;
+
+  public:
+    /// Default constructor
+    StemStopper();
+
+    /** Set the stemmer.
+     *
+     *  @param stem	The Xapian::Stem object to set.
+     */
+    void set_stemmer(const Xapian::Stem &stemmer);
```

Comment:
I think it would make more sense to just take the stemmer as a constructor parameter.  It's not useful to use this class without setting a stemmer, and it's not useful to change the stemmer mid-use, as you'll end up with a mixture of stems from different algorithms.

Also the `@param` doesn't match the actual parameter name.